BlueprintUI/Sources/Layout/LayoutWriter.swift

```diff
@@ -0,0 +1,223 @@
+    /// the environment and sizing of the layout, and the `Builder` itself, which you use to
+    /// add child elements to the layout.
+    public init(_ build : @escaping Build) {
+        self.build = build
+    }
+    
+    /// The builder type passed to the `LayoutWriter` initializer.
+    public typealias Build = (Context, inout Builder) -> ()
+    
+    /// The builder used to create the custom layout.
+    public let build : Build
+    
+    //
+    // MARK: Element
+    //
+    
+    public var content: ElementContent {
+        ElementContent { size, env in
+            var builder = Builder()
+            self.build(Context(size: size, environment: env), &builder)
+            return InnerElement(builder: builder)
+        }
+    }
+    
+    public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
+        nil
+    }
+}
+
+
+extension LayoutWriter {
+    
+    /// The builder is the primary surface area you interact with when using a `LayoutWriter`.
+    ///
+    /// It provides you the ability to manage the sizing and measurement of the final layout,
+    /// alongside methods to add and manage the children of the layout.
+    public struct Builder {
+        
+        //
+        // MARK: Managing Sizing
+        //
+        
+        /// How the size of the layout should be calculated. Defaults to `.unionOfChildren`,
+        /// which means the size will be big enough to contain the frames of all contained children.
+        public var sizing : Sizing = .unionOfChildren
+        
+        //
+        // MARK: Managing Children
+        //
+        
+        /// The children of the custom layout, which specifies the child element and its frame.
+        ///
+        /// Note
+        /// ----
+        /// You rarely need to access this property directly. Instead, add children via
+        /// the various provided `add(...)` methods. However, if you're `map`-ing over an array
+        /// or other collection of content, using this property directly is useful.
+        ///
+        public var children : [Child] = []
+        
+        /// Adds a new child element to the layout with the provided frame.
+        public mutating func add(
+            with frame: CGRect,
+            child : Element
+        ) {
+            self.children.append(.init(frame: frame, element: child))
+        }
+        
+        /// Adds a new child element to the layout with the provided frame.
+        /// The frame is passed to the child provider.
+        public mutating func add(
+            with frame: CGRect,
+            child : (CGRect) -> Element
+        ) {
+            self.add(with: frame, child: child(frame))
+        }
+        
+        /// Adds a new child element to the layout with the provided frame.
+        /// The frame is passed to the child provider.
+        public mutating func add(
+            with frame: () -> CGRect,
+            child : (CGRect) -> Element
```

Comment:
My eventual goal is to begin pushing Blueprint towards the SwiftUI model of taking in closures for child elements, since it gives you a nice scope to do setup work – eg how GeometryReader, Button, etc, take closures for providing their content. But also fine to not do that here until we add it to more core types

Reply:
Removed a couple